CMakeLists.txt

```diff
@@ -128,6 +131,64 @@ if( NOT WITH_REFERENCE_BACKEND AND
 	message( FATAL_ERROR "At least one backend should be enabled")
 endif()
 
+# Adding backends for tests only
+set( AVAILABLE_TEST_BACKENDS "" )
+if( WITH_REFERENCE_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "reference" )
+endif()
+if( WITH_OMP_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "reference_omp" )
+endif()
+if( WITH_HYPERDAGS_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "hyperdags" )
+endif()
+if( WITH_NONBLOCKING_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "nonblocking" )
+endif()
+if( WITH_BSP1D_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "bsp1d" )
+endif()
+if( WITH_HYBRID_BACKEND )
+	list( APPEND AVAILABLE_TEST_BACKENDS "hybrid" )
+endif()
+
+# Enabling reference(+omp) backend if bsp1d is active
+if( WITH_BSP1D_BACKEND )
+	set( WITH_REFERENCE_BACKEND ON )
+	set( WITH_OMP_BACKEND ON )
+	message( STATUS "Enabling compilation of reference backend: required by bsp1d" )
+endif()
+
+# Enabling reference(+omp) backend if hybrid is active
+if( WITH_HYBRID_BACKEND )
+	set( WITH_OMP_BACKEND ON )
```

Comment:
should probably be `WITH_REFERENCE_BACKEND` ?

Reply:
Same as below 
Fixed [here](https://github.com/Algebraic-Programming/ALP/pull/200/commits/5b972d4890323e1b92237b8928a8cb4c468ac038)